courses/advanced-spark/chapters/subprogram_contracts.rst

```diff
@@ -88,26 +94,49 @@ Precondition
 
 .. code:: ada
 
-    function Sqrt (X : Float) return Float with
-      Pre => X >= 0.0 or else raise Argument_Error;
+    package Show_Precondition is
```

Comment:
This looks exactly like the previous Show_Dynamic_Behavior.

Reply:
And it is. :-) It was like that in the original version of the slides. 

I didn't change the original code examples. but rather just created a "complete" version based on them.